### PR TITLE
[boot] E: Split mboot parser function

### LINKED ISSUE #426: [mm] Improve mboot parse function modularity
## Description
The mboot parse function are very long, it must be split in different functions.

## Change Files
- src/mboot/mod.rs

### PR DESCRIPTION
**Pull request by rafael0121**
## Description
The parse methods were separated in different functions, improving modularity and maintainability.

This PR closes #426

### CODE REVIEW

File: src/mboot/mod.rs [hunk 1]
```diff
@@ -258,7 +262,208 @@ impl core::fmt::Debug for MbootTag {
...
+///
+/// # Description
+///
+/// Parse Memory Regions from Multiboot tag.
+///
+/// # Returns
+///
+/// Upon success, returns memory region. Otherwise, it returns an error.
+///
+fn parse_mmap(tag: &MbootTag,
```
Comment (ppenna) [inline src/mboot/mod.rs:317]: Please remove trailling space.

File: src/mboot/mod.rs [hunk 2]
```diff
@@ -258,7 +262,208 @@ impl core::fmt::Debug for MbootTag {
...
+    rsdp.display();
+    let rsdt: *const AcpiSdtHeader = rsdp.rsdt_addr as *const AcpiSdtHeader;
+    let rsdt: AcpiSdtHeader = unsafe { AcpiSdtHeader::from_ptr(rsdt) }?;
+    rsdt.display();
+
+    let ptr: *const AcpiSdtHeader = unsafe {
+        cpu::acpi::find_table_by_sig(rsdp.rsdt_addr as *const AcpiSdtHeader, "APIC")?
+    };
+    rsdp.display();
+    let madt = match unsafe { Madt::parse(ptr as *const Madt) } {
```
Comment (ppenna) [inline src/mboot/mod.rs:400]: Please add type annotation.

File: src/mboot/mod.rs [hunk 3]
```diff
@@ -258,7 +262,208 @@ impl core::fmt::Debug for MbootTag {
 }
 
 //==================================================================================================
-// Implementations
+// Parse Functions
+//==================================================================================================
```
Comment (ppenna) [inline src/mboot/mod.rs:266]: Just move all standalone functions to the "Standalone Functions" section.

File: src/mboot/mod.rs [hunk 4]
```diff
@@ -258,7 +262,208 @@ impl core::fmt::Debug for MbootTag {
...
+
+///
+/// # Description
+///
+/// Parse modules from Multiboot tag.
+///
```
Comment (ppenna) [inline src/mboot/mod.rs:272]: Please add a documentation for the parameters.

File: src/mboot/mod.rs [hunk 5]
```diff
@@ -258,7 +262,208 @@ impl core::fmt::Debug for MbootTag {
...
+    kernel_modules.push_back(module);
+
+    Ok(kernel_modules)
+}
+
+///
+/// # Description
+///
+/// Parse Memory Regions from Multiboot tag.
+///
```
Comment (ppenna) [inline src/mboot/mod.rs:312]: Please add a documentation for parameters.

File: src/mboot/mod.rs [hunk 6]
```diff
@@ -258,7 +262,208 @@ impl core::fmt::Debug for MbootTag {
...
+    }
+
+    Ok(memory_regions)
+}
+
+///
+/// # Description
+///
+/// Parse Old Root System Description Pointer (RSDP) from Multiboot tag.
+///
```
Comment (ppenna) [inline src/mboot/mod.rs:375]: Please add a documentation for paramemeters.

File: src/mboot/mod.rs [hunk 7]
```diff
@@ -258,7 +262,208 @@ impl core::fmt::Debug for MbootTag {
...
+        },
+    };
+
+    Ok(madt)
+}
+
+///
+/// # Description
+///
+/// Parse New Root System Description Pointer (RSDP) from Multiboot tag.
```
Comment (ppenna) [inline src/mboot/mod.rs:448]: Please add a documentation for parameters.

File: src/mboot/mod.rs [hunk 8]
```diff
@@ -258,7 +262,208 @@ impl core::fmt::Debug for MbootTag {
...
+    };
+
+    Ok(acpi)
+}
+
+//==================================================================================================
+// Standalone Functions
 //==================================================================================================
 
 pub fn parse(bootloader_magic: u32, addr: usize) -> Result<BootInfo, Error> {
```
Comment (ppenna) [inline src/mboot/mod.rs:469]: Please add a documentation for this function.

File: src/mboot/mod.rs (hunk 1 above)
Comment (rafael0121) [reply src/mboot/mod.rs:317]: Done.

File: src/mboot/mod.rs (hunk 2 above)
Comment (rafael0121) [reply src/mboot/mod.rs:400]: Done.

File: src/mboot/mod.rs (hunk 3 above)
Comment (rafael0121) [reply src/mboot/mod.rs:266]: Done.

File: src/mboot/mod.rs (hunk 7 above)
Comment (rafael0121) [reply src/mboot/mod.rs:448]: Done.

File: src/mboot/mod.rs (hunk 5 above)
Comment (rafael0121) [reply src/mboot/mod.rs:312]: Done.

File: src/mboot/mod.rs (hunk 4 above)
Comment (rafael0121) [reply src/mboot/mod.rs:272]: Done.

File: src/mboot/mod.rs (hunk 8 above)
Comment (rafael0121) [reply src/mboot/mod.rs:469]: Done.

**Review by ppenna — APPROVED** [review]
LGTM

**Pull request merged**